ccdproc/image_collection.py

```diff
@@ -652,10 +663,17 @@ def _generator(self, return_type,
 
             file_name = path.basename(full_path)
 
+            if 'unit' not in ccd_kwargs:
```

Comment:
If unit is ``None`` then ``_generator`` is always going to throw ``ValueError: a unit for CCDData must be specified.`` even if I just would like to use ``headers`` or ``data``, for instance.